From 7a2e459dcd0312198189807e5464873288ca4491 Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Thu, 24 Oct 2013 18:37:00 +0700
Subject: Dalvik: insertPhiNodeOperands should do a null check

BZ: 144514

insertPhiNodeOperands utility should expect that
predBB->dataFlowInfo->dalvikToSSAMapExit can be null
if pred is kFromInterpreter basic block.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I1f99f2d52ca7732ee84ab38d29018af50010bbc8
Orig-MCG-Change-Id: I880fce9efe1b6b8c4f39672e549fa301f67ebb33
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/SSATransformation.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/vm/compiler/SSATransformation.cpp b/vm/compiler/SSATransformation.cpp
index fee2a52..013a76f 100644
--- a/vm/compiler/SSATransformation.cpp
+++ b/vm/compiler/SSATransformation.cpp
@@ -677,6 +677,13 @@ static bool insertPhiNodeOperands(CompilationUnit *cUnit, BasicBlock *bb)
             //Paranoid
             assert (predBB != 0);
 
+            //If dalvikToSSAMapExit is not created than we did not traverse this bb in SSA conversion
+            //So it cannot give us anything interesting then ingore it
+            if (predBB->dataFlowInfo->dalvikToSSAMapExit == 0)
+            {
+                continue;
+            }
+
             //Now get the SSA combination (SSA register, SSA Subscript) via the SSA map at the exit of the predecessor
             int encodedSSAValue =
                 predBB->dataFlowInfo->dalvikToSSAMapExit[dalvikReg];
-- 
1.7.4.1

